### PR TITLE
8: Expand fgs system tests

### LINKED ISSUE #8: Add more system tests for fast grid scan
Following the fix of s03 (https://gitlab.diamond.ac.uk/controls/python3/s03_utils/-/merge_requests/4) the system tests for the fast grid scan device should be expanded to include the kickoff and completion.

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes #8 

Note, is branched from https://github.com/DiamondLightSource/python-artemis/pull/7 so merge that PR first.

To test:
* Ensure you have https://gitlab.diamond.ac.uk/controls/python3/s03_utils/-/merge_requests/4 running for the fast grid scan (this is not yet on ws019
* Run the system tests and confirm they all pass

### CODE REVIEW

File: src/artemis/devices/fast_grid_scan.py
Comment (bentom08) [inline src/artemis/devices/fast_grid_scan.py:185]: If you wait until my eiger code is merged, there's a util included (await_value in status.py) that will create this object for you and let you put the wait on the same line - await_value(pv, value).wait(timeout)

File: src/artemis/devices/fast_grid_scan.py
Comment (bentom08) [inline src/artemis/devices/fast_grid_scan.py:198]: you can use set_and_wait in utils.epics_pvs for this, or just add the wait onto the end of the set rather than creating the status object

File: src/artemis/devices/system_tests/test_gridscan_system.py
Comment (bentom08) [inline src/artemis/devices/system_tests/test_gridscan_system.py:28]: this seems a little superfluous on top of the previous test

**Review by bentom08 — CHANGES_REQUESTED** [review]
overall looks good, just a few minor comments

File: src/artemis/devices/system_tests/test_gridscan_system.py
Comment (DominicOram) [reply src/artemis/devices/system_tests/test_gridscan_system.py:28]: Agreed, I think I was building the tests up incrementally and so had the duplicates in there. If they were fast unit tests I might argue that it's useful to have explicit separate ones but as they'll take some time to run we should keep the number of tests low.

**Comment by DominicOram** [comment]
Thanks for the review, changes made. I also found that despite the new changes being on s03 the tests still fail. I would normally look into fixing it but i03 testing takes priority so have fudged it and created https://github.com/DiamondLightSource/python-artemis/issues/13 to fix properly

**Review by bentom08 — APPROVED** [review]
looks good

**Pull request merged**
